cdap-app-fabric/src/main/java/co/cask/cdap/app/guice/DistributedProgramRunnableModule.java

```diff
@@ -174,7 +174,7 @@ protected void configure() {
           // For binding DataSet transaction stuff
           install(new DataFabricFacadeModule());
 
-          bind(ProgramStateWriter.class).to(DirectStoreProgramStateWriter.class);
+          bind(ProgramStateWriter.class).to(MessagingProgramStateWriter.class);
```

Comment:
I have tried to remove this binding but I got a bunch of injector issues. We still do as it's needed in `AbstractProgramRuntimeService` where we write the starting state for all programs.